### PR TITLE
Add 2023 Washington Tax Rules

### LINKED ISSUE #4201: Add 2023 Washington Tax Rules
No income tax in Washington state.
https://dor.wa.gov/taxes-rates/income-tax
https://app.leg.wa.gov/RCW/default.aspx

### PR DESCRIPTION
**Pull request by Mshuning**
Fixes #4201

### CODE REVIEW

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
   unit: currency-USD
   period: year
   reference:
-    - title: "RCW 82.87.080(2) Charitable donation deduction"
+    - title: RCW 82.87.080(2) Charitable donation deduction
       href: https://app.leg.wa.gov/RCW/default.aspx?cite=82.87.080
+    - title: Capital gains tax | Washington Department of Revenue
+      href: https://dor.wa.gov/taxes-rates/other-taxes/capital-gains-tax
 values:
   0000-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:12]: can we remove the 0 values?

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
   2022-01-01: 100_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:13]: check for 2021 values & applicability - if not in effect in 2021 - make an in_effect bool parameter

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/amount.yaml [hunk 3]
```diff
@@ -1,28 +1,33 @@
 description: Washington Working Families Tax Credit maximum amount by number of children.
 metadata:
   label: WA Working Families Tax Credit maximum amount by number of children
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/amount.yaml:3]:
```suggestion
  label: Washington Working Families Tax Credit maximum amount by number of children
```

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/amount.yaml:3]: everywhere

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 metadata:
   label: Washington capital gains charitable contribution cap
-  name: wa_ltcg_charitable_contributions_cap
   unit: currency-USD
   period: year
   reference:
-    - title: "RCW 82.87.080(2) Charitable donation deduction"
+    - title: RCW 82.87.080(2) Charitable donation deduction
       href: https://app.leg.wa.gov/RCW/default.aspx?cite=82.87.080
+    - title: Capital gains tax | Washington Department of Revenue
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:9]: could you add 2022 form as well

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
+      href: https://dor.wa.gov/taxes-rates/other-taxes/capital-gains-tax
+    - title: 2021 Tax Legislation | Washington Department of Revenue
+      href: https://dor.wa.gov/forms-publications/publications-subject/tax-topics/2021-tax-legislation
 values:
-  0000-01-01: 0
-  2022-01-01: 100_000
+  2021-01-01: 100_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:14]: is it uprated with inflation?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:14]: same for all yaml

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/in_effect.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+description: Washington Working Families Tax Credit applies if this is true.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/in_effect.yaml:1]:
```suggestion
description: Washington applies the Working Families Tax Credit applies if this is true.
```

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/in_effect.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+values:
+  0000-01-01: false
+  2022-01-01: true
+metadata:
+  unit: bool
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/in_effect.yaml:6]:
```suggestion
  unit: bool
  period: year
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml (hunk 5 above)
Comment (Mshuning) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:14]: https://lawfilesext.leg.wa.gov/biennium/2021-22/Pdf/Bills/Session%20Laws/Senate/5096-S.SL.pdf#page=17

> Beginning December 2023 and each December thereafter, the department must adjust the applicable amounts by multiplying the current applicable amounts by one plus the percentage by which the most current consumer price index available on December 1st of the current year exceeds the consumer price index for the prior 12-month period, and rounding the result to the nearest $1,000.

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml (hunk 4 above)
Comment (Mshuning) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:9]: couldn't find any 2022 form

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:9]: Is there a tax form?

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:16]: Please add uprating metadata with rounding

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:16]: for all yaml

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml (hunk 4 above)
Comment (Mshuning) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/cap.yaml:9]: seems no 
https://dor.wa.gov/manage-business/my-dor-help/capital-gains-my-dor-help#filereturn

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/amount.yaml [hunk 8]
```diff
@@ -1,28 +1,33 @@
...
   - threshold:
       2022-01-01: 0
     amount:
       2022-01-01: 300
+      2023-01-01: 315
   - threshold:
       2022-01-01: 1
     amount:
       2022-01-01: 600
+      2023-01-01: 625
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/amount.yaml:23]: does this change with inflation as well?

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/standard.yaml [hunk 9]
```diff
@@ -1,12 +1,22 @@
...
-    - title: "RCW 82.87.060(1) Deductions"
+    - title: RCW 82.87.060(1) Deductions
       href: https://app.leg.wa.gov/RCW/default.aspx?cite=82.87.060
+    - title: Capital gains tax | Washington Department of Revenue
+      href: https://dor.wa.gov/taxes-rates/other-taxes/capital-gains-tax
+    - title: 2021 Tax Legislation | Washington Department of Revenue
+      href: https://dor.wa.gov/forms-publications/publications-subject/tax-topics/2021-tax-legislation#Working%20families%20tax%20credit
+    - title: 2021 Regular Session CAPITAL GAINS TAX - Sec.7.(1)
+      href: https://lawfilesext.leg.wa.gov/biennium/2021-22/Pdf/Bills/Session%20Laws/Senate/5096-S.SL.pdf#page=9
+  uprating:
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/standard.yaml:15]: This implies that the parameter is adjusted every year (with inflation) so we should have a different 2022 amount

Comment (Mshuning) [reply policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/standard.yaml:15]: Should I change 2021 amount into 2022?
> Effective July 25, 2021; however, the new tax is not imposed until Jan. 1, 2022 ([ESSB 5096](http://lawfilesext.leg.wa.gov/biennium/2021-22/Pdf/Bills/Session%20Laws/Senate/5096-S.SL.pdf?q=20210517144444), Chapter 196, Laws of 2021).

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/amount.yaml (hunk 8 above)
Comment (Mshuning) [reply policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/amount.yaml:23]: seems no

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/exemption.yaml [hunk 10]
```diff
@@ -1,12 +1,17 @@
...
+    - title: Capital gains tax | Washington Department of Revenue
+      href: https://dor.wa.gov/taxes-rates/other-taxes/capital-gains-tax
+    - title: 2021 Tax Legislation | Washington Department of Revenue
+      href: https://dor.wa.gov/forms-publications/publications-subject/tax-topics/2021-tax-legislation
+    - title: 2021 Regular Session CAPITAL GAINS TAX - Sec.9.(1)
+      href: https://lawfilesext.leg.wa.gov/biennium/2021-22/Pdf/Bills/Session%20Laws/Senate/5096-S.SL.pdf#page=11
 values:
-  0000-01-01: 0
-  2022-01-01: 250_000
+  2021-01-01: 250_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/deductions/charitable/exemption.yaml:16]: change all start dates to 2022

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/in_effect.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/in_effect.yaml:1]: lets move the in_effect to general wa_tax_income file and apply it to the capital gains as well

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/rate.yaml [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
   label: Washington capital gains tax rate
-  name: wa_ltcg_rate
   unit: /1
   period: year
   reference:
-    - title: "RCW 82.87.040(1) Tax imposed—Long-term capital assets."
+    - title: RCW 82.87.040(1) Tax imposed—Long-term capital assets.
       href: https://app.leg.wa.gov/RCW/default.aspx?cite=82.87.040
+    - title: Capital gains tax | Washington Department of Revenue
+      href: https://dor.wa.gov/taxes-rates/other-taxes/capital-gains-tax#Background
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/capital_gains/rate.yaml:10]: image

Similar language should be present for all the values that are adjusted annually

File: policyengine_us/variables/gov/states/wa/tax/income/credits/wa_working_families_tax_credit.py [hunk 12]
```diff
@@ -11,35 +11,38 @@ class wa_working_families_tax_credit(Variable):
     defined_for = StateCode.WA
 
     def formula_2022(tax_unit, period, parameters):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/wa/tax/income/credits/wa_working_families_tax_credit.py:13]:
```suggestion
    def formula(tax_unit, period, parameters):
```

File: policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml [hunk 13]
```diff
@@ -0,0 +1,25 @@
+description: Washington applies its income tax rules if this is true.  
+values:
+  0000-01-01: false
+  2022-01-01: true
+metadata:
+  unit: bool
+  period: year  
+  label: Washington Income Tax Rules in effect
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml:8]:
```suggestion
  label: Washington income tax rules in effect
```

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start_below_eitc.yaml [hunk 14]
```diff
@@ -1,13 +1,12 @@
 description: Washington Working Families Tax Credit phase-out start, below the EITC's income limit, by number of children.
 metadata:
-  label: WA Working Families Tax Credit phase-out start, below the EITC's income limit, by number of children
-  name: wa_wftc_phase_out_start_below_eitc_income_limit
+  label: Washington Working Families Tax Credit phase-out start, below the EITC's income limit, by number of children
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start_below_eitc.yaml:3]:
```suggestion
  label: Washington Working Families Tax Credit phase-out start
```

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start_below_eitc.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start_below_eitc.yaml:1]: can we just name this file start.yaml (are there other phase out starts?)

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/min_amount.yaml [hunk 15]
```diff
@@ -1,11 +1,10 @@
...
-  name: wa_wftc_min_amount
+  label: Washington Working Families Tax Credit minimum amount
   unit: currency-USD
   period: year
   reference:
-    - title: "RCW 82.08.0206 Exemptions—Working families—Eligible low-income persons—Penalties. (3)(c)"
+    - title: RCW 82.08.0206 Exemptions—Working families—Eligible low-income persons—Penalties. (3)(c)
       href: https://app.leg.wa.gov/RCW/default.aspx?cite=82.08.0206
 values:
   2022-01-01: 50
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/min_amount.yaml:10]: no tax form reference?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/min_amount.yaml (hunk 15 above)
Comment (Mshuning) [reply policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/min_amount.yaml:10]: didn't find any
seems that they use website to fill tax form

File: policyengine_us/variables/gov/states/wa/tax/income/wa_capital_gains_tax.py [hunk 16]
```diff
@@ -10,23 +10,26 @@ class wa_capital_gains_tax(Variable):
...
+            charitable_deduction = min_(
+                charitable_contributions_above_exemption,
+                p.deductions.charitable.cap,
+            )
+            # Get base after standard and charitable deductions.
+            taxable_ltcg = max_(
+                0, ltcg - charitable_deduction - p.deductions.standard
+            )
+            return taxable_ltcg * p.rate
+        return 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/wa/tax/income/wa_capital_gains_tax.py:35]: add a test for 2021 for each file so we pass the test coverage check

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start.yaml:1]: why the rename? this is not a standard phase-out

File: policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml:1]: don't we typically do this by making a list parameter of the applicable credits by year?

**Review by MaxGhenis — COMMENTED** [review]
generally curious to the rationale of the refactor. would suggest adding a wftc eligible variable

File: policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml:1]: Yes, I think that would be the test approach, I generally am opposed to the `formula_x` structure as this limits our ability to adjust and display intermediate variables, in this case we should create a list parameter for credits in DC.

This is the first time that we have a state income tax structure only be applied from 2022 on (no capital gains tax before 2022 in WA and no other income tax ever) and we were thinking on how to represent that

File: policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/credits/working_families_tax_credit/phase_out/start.yaml:1]: Just felt bulky but I see the use case - changed back

File: policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml:1]: Changes to a list for credit but still have the in effect for the main tax var to not break the nat income tree pre-2022

Comment (Mshuning) [reply policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml:1]: how about wa_capital_gains_tax?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/wa/tax/income/in_effect.yaml:1]: I think we keep that the way it is in this PR and adjust in a separate issue as we would need to make adjustments to a lot of the crucial state income tax lists which I would like to isolate

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**